src/Agent.Worker/TaskManager.cs

```diff
@@ -450,14 +451,13 @@ public HandlerData()
             Inputs = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         }
 
-        public bool PreferredOnCurrentPlatform()
+        public bool PreferredOnPlatform(PlatformUtil.OS os)
         {
-#if OS_WINDOWS
-            const string CurrentPlatform = "windows";
-            return Platforms?.Any(x => string.Equals(x, CurrentPlatform, StringComparison.OrdinalIgnoreCase)) ?? false;
-#else
+            if (os == PlatformUtil.OS.Windows)
+            {
+                return Platforms?.Any(x => string.Equals(x, os.ToString().ToLowerInvariant(), StringComparison.OrdinalIgnoreCase)) ?? false;
```

Comment:
Do you need ToLowerInvariant() here? The comparer should do the work for you.

Reply:
weird, that was from Matt's change. I'll fix this.